### ISSUE TITLE
PyTorch example, mutiple workers, cuda initialization error.

### ISSUE THREAD

**Issue by sheep94lion** [kubeflow/kfserving#851]
/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]
I am following the TFServing examples for PyTorch.
https://github.com/kubeflow/kfserving/tree/master/docs/samples/pytorch
I use this command to start up the server:
python -m pytorchserver --model_dir ./ --model_name pytorchmodel --model_class_name Net
By default, in `python/pytorch_server/pytorch_server/__main__.py`:
    kfserving.KFServer().start([model])
`workers` is not specified and defaults to the number of cores as in `tornado.tcpserver.py`.

When I proceed with the pytorch example to test the inference server:
```
import torch
import torchvision
import torchvision.transforms as transforms
import requests

transform = transforms.Compose([transforms.ToTensor(),
                                transforms.Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5))])
testset = torchvision.datasets.CIFAR10(root='./data', train=False,
                                       download=True, transform=transform)
testloader = torch.utils.data.DataLoader(testset, batch_size=4,
                                         shuffle=False, num_workers=2)
dataiter = iter(testloader)
images, labels = dataiter.next()
formData = {
    'instances': images[0:1].tolist()
}
res = requests.post('http://localhost:8080/v1/models/pytorchmodel:predict', json=formData)
print(res)
print(res.text)
```
Error occurs:
```
(pytorch13) yizhao@pop-os:~/kfserving$ /home/yizhao/anaconda3/envs/pytorch13/bin/python /home/yizhao/kfserving/python/pytorchserver/pytorchserver/__main__.py
[I 200530 11:19:11 storage:35] Copying contents of /home/yizhao/kfserving/docs/samples/pytorch to local
[I 200530 11:19:13 kfserver:88] Registering model: pytorchmodel
[I 200530 11:19:13 kfserver:77] Listening on port 8081
[I 200530 11:19:13 kfserver:79] Will fork 0 workers
[I 200530 11:19:13 process:126] Starting 12 processes
[E 200530 11:19:22 web:1788] Uncaught exception POST /v1/models/pytorchmodel:predict (::1)
    HTTPServerRequest(protocol='http', host='localhost:8081', method='POST', uri='/v1/models/pytorchmodel:predict', version='HTTP/1.1', remote_ip='::1')
    Traceback (most recent call last):
      File "/home/yizhao/kfserving/python/pytorchserver/pytorchserver/model.py", line 67, in predict
        inputs = torch.tensor(request["instances"]).to(self.device)
    RuntimeError: CUDA error: initialization error
    
    During handling of the above exception, another exception occurred:
    
    Traceback (most recent call last):
      File "/home/yizhao/anaconda3/envs/pytorch13/lib/python3.7/site-packages/tornado/web.py", line 1697, in _execute
        result = method(*self.path_args, **self.path_kwargs)
      File "/home/yizhao/anaconda3/envs/pytorch13/lib/python3.7/site-packages/kfserving/handlers/http.py", line 44, in post
        response = model.predict(request)
      File "/home/yizhao/kfserving/python/pytorchserver/pytorchserver/model.py", line 70, in predict
        "Failed to initialize Torch Tensor from inputs: %s, %s" % (e, inputs))
    TypeError: Failed to initialize Torch Tensor from inputs: CUDA error: initialization error, []
[E 200530 11:19:22 web:2246] 500 POST /v1/models/pytorchmodel:predict (::1) 7.10ms
```
Then I change `workders` to 1: `kfserving.KFServer(workers=1).start([model])`, it works fine:
```
(pytorch13) yizhao@pop-os:~/kfserving$ /home/yizhao/anaconda3/envs/pytorch13/bin/python /home/yizhao/kfserving/python/pytorchserver/pytorchserver/__main__.py
[I 200530 11:20:48 storage:35] Copying contents of /home/yizhao/kfserving/docs/samples/pytorch to local
[I 200530 11:20:50 kfserver:88] Registering model: pytorchmodel
[I 200530 11:20:50 kfserver:77] Listening on port 8081
[I 200530 11:20:50 kfserver:79] Will fork 1 workers
[I 200530 11:20:56 web:2246] 200 POST /v1/models/pytorchmodel:predict (::1) 113.72ms
```
I also try `workers=2`, error occur:
```
(pytorch13) yizhao@pop-os:~/kfserving$ /home/yizhao/anaconda3/envs/pytorch13/bin/python /home/yizhao/kfserving/python/pytorchserver/pytorchserver/__main__.py
[I 200530 11:22:35 storage:35] Copying contents of /home/yizhao/kfserving/docs/samples/pytorch to local
[I 200530 11:22:37 kfserver:88] Registering model: pytorchmodel
[I 200530 11:22:37 kfserver:77] Listening on port 8081
[I 200530 11:22:37 kfserver:79] Will fork 2 workers
[I 200530 11:22:37 process:126] Starting 2 processes
[E 200530 11:22:43 web:1788] Uncaught exception POST /v1/models/pytorchmodel:predict (::1)
    HTTPServerRequest(protocol='http', host='localhost:8081', method='POST', uri='/v1/models/pytorchmodel:predict', version='HTTP/1.1', remote_ip='::1')
    Traceback (most recent call last):
      File "/home/yizhao/kfserving/python/pytorchserver/pytorchserver/model.py", line 67, in predict
        inputs = torch.tensor(request["instances"]).to(self.device)
    RuntimeError: CUDA error: initialization error
    
    During handling of the above exception, another exception occurred:
    
    Traceback (most recent call last):
      File "/home/yizhao/anaconda3/envs/pytorch13/lib/python3.7/site-packages/tornado/web.py", line 1697, in _execute
        result = method(*self.path_args, **self.path_kwargs)
      File "/home/yizhao/anaconda3/envs/pytorch13/lib/python3.7/site-packages/kfserving/handlers/http.py", line 44, in post
        response = model.predict(request)
      File "/home/yizhao/kfserving/python/pytorchserver/pytorchserver/model.py", line 70, in predict
        "Failed to initialize Torch Tensor from inputs: %s, %s" % (e, inputs))
    TypeError: Failed to initialize Torch Tensor from inputs: CUDA error: initialization error, []
[E 200530 11:22:43 web:2246] 500 POST /v1/models/pytorchmodel:predict (::1) 6.95ms
```
Is this a bug? or I did something wrong? or my laptop just does not have enough resources to run multiple workers? My laptop has 6c/12t cpu, 16g RAM. and GTX 1050Ti max q with 4G memory.
**What did you expect to happen:**
Multiple workers also work fine.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Istio Version:
- Knative Version:
- KFServing Version:
- Kubeflow version:
- Kfdef:[k8s_istio/istio_dex/gcp_basic_auth/gcp_iap/aws/aws_cognito/ibm]
- Minikube version:
- Kubernetes version: (use `kubectl version`):1.18
- OS (e.g. from `/etc/os-release`):pop os 20.04

**Comment by yuzisun** [comment]
@sheep94lion maybe related to this issue  https://github.com/pytorch/pytorch/issues/334?

**Comment by yuzisun** [comment]
for pytorchserver GPU now we default to 1 worker which is released in 0.5.
